python/cuml/manifold/umap.pyx

```diff
@@ -384,25 +394,30 @@ class UMAP(Base):
         state["target_weights"] = umap_params.target_weights
         state["target_metric"] = umap_params.target_metric
 
-        del state["umap_params"]
+        if not self.MNMG:
+            del state["umap_params"]
 
         return state
 
     def __del__(self):
         cdef UMAPParams * umap_params
         if hasattr(self, 'umap_params'):
-            umap_params = <UMAPParams*><size_t>self.umap_params
-            free(umap_params)
+            if self.umap_params != 0:
+                umap_params = <UMAPParams*><size_t>self.umap_params
+                free(umap_params)
 
     def __setstate__(self, state):
-        super(UMAP, self).__init__(handle=None, verbose=state['verbose'])
 
-        if "X_m" in state and state["X_m"] is not None:
-            # fit has not yet been called
-            state["X_m"] = row_matrix(state["X_m"])
-            state["embedding_"] = row_matrix(state["embedding_"])
+        if not self.MNMG:
+            super(UMAP, self).__init__(handle=None, verbose=state['verbose'])
+
+            if "X_m" in state and state["X_m"] is not None:
+                # fit has not yet been called
+                state["X_m"] = row_matrix(state["X_m"])
```

Comment:
Ideally, we should be using `input_to_cuml_array` here. Eventually, this `row_matrix` function will go away entirely.